### PR TITLE
implement custom copy for avail dict in pkg

### PR DESCRIPTION
**Pull request by KristofferC**
Shaves off ~0.3 seconds from `Pkg.resolve()`

### CODE REVIEW

**Comment by KristofferC** [comment]
Will add test that hash of dict is the same in a while.

**Comment by KristofferC** [comment]
Test added

File: base/pkg/query.jl [hunk 1]
```diff
@@ -27,8 +27,21 @@ function requirements(reqs::Dict, fix::Dict, avail::Dict)
     reqs
 end
 
+# Specialized copy for the avail argument below because the deepcopy is slow
+function availcopy(avail)
+    new_avail = similar(avail)
+    for (pkg, vers_avail) in avail
+        new_vers_avail = similar(vers_avail)
+        for (version, avail) in vers_avail
```
Comment (martinholters) [inline base/pkg/query.jl]: Maybe not reuse `avail` here?

Comment (KristofferC) [inline base/pkg/query.jl]: yeah, good point!

**Comment by KristofferC** [comment]
Could potentially be backported after getting battle tested on master for a while.

**Comment by KristofferC** [comment]
Ok to merge?

**Pull request merged**